src/services/tx/generateSafeTxHash.ts

```diff
@@ -0,0 +1,40 @@
+import { utils } from 'ethers'
+import { generateTypedData } from '@gnosis.pm/safe-core-sdk-utils'
+import type { SafeTransactionData } from '@gnosis.pm/safe-core-sdk-types'
+import type { SafeInfo } from '@gnosis.pm/safe-react-gateway-sdk'
+
+import { isValidSafeVersion } from '@/hooks/coreSDK/safeCoreSDK'
+import ErrorCodes from '../exceptions/ErrorCodes'
+import { logError } from '../exceptions'
+
+export const generateSafeTxHash = (safe: SafeInfo, data: SafeTransactionData): string => {
+  if (!isValidSafeVersion(safe.version)) {
+    return ''
```

Comment:
Should this not also be logged in some way as an error?

Or should this method probably return `undefined` if the hash creation fails? Currently when we call this we do not check if the returned value is an empty string.

Reply:
> Should this not also be logged in some way as an error?
> 
> Or should this method probably return `undefined` if the hash creation fails? Currently when we call this we do not check if the returned value is an empty string.

`isValidSafeVersion` should perhaps be an "invariant" function but I think we should revisit this in a separate issue.

I have now changed `generateSafeTxHash` to return `undefined` if the Safe version is invalid/generation is unsuccessfuly. `useTxNotifications` also only shows the `safeTxHash` if it is defined.